docs/architecture/ADR-004-interest-and-reserves.md

```diff
@@ -153,25 +126,16 @@ for _, coin := range newReserves {
 
 ### uToken Exchange Rate Impact
 
-In financial terms, accrued interest is split between reserves and lender profits.
-
-The token:uToken exchange rate, which would have previously been calculated by `(total tokens borrowed + module account balance) / uTokens in circulation` for a given token type and associated uToken denomination, must now be computed as `(total tokens borrowed + module account balance - reserved amount) / uTokens in circulation`.
+The token:uToken exchange rate, which would have previously been calculated by `(total tokens borrowed + module account balance) / uTokens in circulation` for a given token and associated uToken, must now be computed as `(total tokens borrowed + module account balance - reserved amount) / uTokens in circulation`.
 
-Also, because Lend, Withdraw, Borrow, and Repay transactions don't affect the token:uToken exchange rate, the exchange rate only needs to be calculated for each token every `InterestEpoch`, and stored so it can be used during transactions.
+Also, because Lend, Withdraw, Borrow, and Repay transactions don't affect the token:uToken exchange rate, it's enough to update the exchange rate during the EndBlocker `AccrueAllInterest`.
 
-The exchange rate's keeper prefix is constructed as follows:
+The exchange rate is stored under the following key:
 
-```go
-    // exchangeRatePrefix | denom | 0
-    var key []byte
-    key = append(key, KeyPrefixExchangeRate...)
-    key = append(key, []byte(tokenDenom)...)
-    return append(key, 0) // append 0 for null-termination
 ```
-
-### Message Types
-
-No new message types (outside of governance of the many reserve parameters) are required to implement dynamic interest rate and reserve functionality.
+// append 0 for null-termination
+exchangeRatePrefix | denom | 0
+```
```

Comment:
Needs to be removed since exchange rate no longer stored

Reply:
this was already there, I moved it to a common place.
We need to store somewhere the exchange rates.

Reply:
Exchange rate is currently not stored in state - it is computed on demand when needed for queries/transactions.

Reply:
OK, so let's remove it.